### PR TITLE
Recommendations

### PR DESCRIPTION
**Pull request by clopezr9**

### CODE REVIEW

File: mm_evaluation/views.py [hunk 1]
```diff
@@ -172,10 +173,11 @@ class PreviousResults(LoginRequiredMixin, ListView):
 
     def get_queryset(self):
         self.pyme = get_object_or_404(PYME, user=self.request.user)
-        return Autoevaluation.objects.filter(pyme=self.pyme).order_by('last_time_edition')
+        return Autoevaluation.objects.filter(pyme_id=self.pyme).order_by('last_time_edition')
```
Comment (jandrovins) [inline mm_evaluation/views.py:176]: This should not work. `Autoevaluation` instances does not have a `pyme_id` attribute. After `models_adaptations` it is named `pyme`.

File: mm_evaluation/views.py [hunk 2]
```diff
@@ -172,10 +173,11 @@ class PreviousResults(LoginRequiredMixin, ListView):
...
 
       
 class ResultDetail(LoginRequiredMixin, DetailView):
+    """This view inherits from LoginRequiredMixin and DetailView. Shows a graph of the result of every macroprocess in the autoevaluation, also have a query of every macroprocess objec available and a query of the object of GeneralPractice to show the maturity level of the PYME an send it to the template by using context"""
```
Comment (jandrovins) [inline mm_evaluation/views.py:180]: The convention for documentation is Google's: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

File: mm_evaluation/views.py [hunk 3]
```diff
@@ -208,6 +220,40 @@ def get_context_data(self, **kwargs):
...
+        #General practice object
+        maturity_level = GeneralPractice.objects.get(score=lowest_score)
+        context['maturity_level']=maturity_level
+
+        if (lowest_score<5):
+            lowest_score += 1
+            general_recommendation = GeneralPractice.objects.get(score=lowest_score)
+            context['general_recommendation']= general_recommendation
+        else:
+            context['general_recommendation']= "SIGUE ASÍ"
```
Comment (jandrovins) [inline mm_evaluation/views.py:256]: Is there a reason why this is in uppercase?

File: mm_evaluation/views.py [hunk 4]
```diff
@@ -217,6 +263,165 @@ def get(self, request, *args, **kwargs):
             return redirect(reverse('mm_evaluation:denied_access'))
         return response_class
 
+
+class SpecificRecommendationsDetail(DetailView):
+    """This view inherits from LoginRequiredMixin and DetailView. Shows the next level of every SpecificPractice of every macroprocess to use as a recommendation"""
```
Comment (jandrovins) [inline mm_evaluation/views.py:268]: Remember to use Google's convention: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

File: mm_evaluation/views.py [hunk 5]
```diff
@@ -217,6 +263,165 @@ def get(self, request, *args, **kwargs):
...
+
+    # For use in LoginRequiredMixin
+    login_url = reverse_lazy('mm_evaluation:login')
+    permission_denied_message = "Debes ingresar a tu cuenta para acceder a esta sección."
+    
+    template_name = 'mm_evaluation/specificrecommendation.html'
+    model = Autoevaluation
+
+    def get(self, request, pk, ev_pk, *args, **kwargs):
```
Comment (jandrovins) [inline mm_evaluation/views.py:277]: This method seems very complex. Would you mind explaining in comments a little bit?

File: mm_evaluation/views.py (hunk 3 above)
Comment (clopezr9) [reply mm_evaluation/views.py:264]: Nope

File: diagnospymes/settings.py [hunk 6]
```diff
@@ -57,7 +57,7 @@
 TEMPLATES = [
     {
         'BACKEND': 'django.template.backends.django.DjangoTemplates',
-        'DIRS': [],
+        'DIRS': [os.path.join(BASE_DIR, 'templates')],
```
Comment (jandrovins) [inline diagnospymes/settings.py:60]: Is this necessary?

File: mm_evaluation/models.py [hunk 7]
```diff
@@ -43,6 +43,24 @@
         ]
 """Valid education level for PYME's contact."""
 
+"""Valid ID type for PYME's contact. """
```
Comment (jandrovins) [inline mm_evaluation/models.py:46]: The comments should go below the end of the list.

File: mm_evaluation/models.py [hunk 8]
```diff
@@ -43,6 +43,24 @@
...
+VALID_ID_TYPE = [
+        ('CC', 'Cédula de Ciudadanía'),
+        ('CE', 'Cédula de Extranjería'),
+        ('PA', 'Pasaporte'),
+        ]
+
+"""Valid education level for PYME's contact. """
```
Comment (jandrovins) [inline mm_evaluation/models.py:53]: Same as above. Documentation below the end of the list. :dango:

File: mm_evaluation/templates/registration/login.html [hunk 9]
```diff
@@ -1,5 +1,6 @@
 {% if form.errors %}
 <p>Tu usuario o tu contraseña son incorrectos. Intenta de nuevo.</p>
+
```
Comment (jandrovins) [inline mm_evaluation/templates/registration/login.html:3]: This empty line should not be here.

File: mm_evaluation/views.py [hunk 10]
```diff
@@ -180,10 +181,13 @@ class PreviousResults(LoginRequiredMixin, ListView):
 
     def get_queryset(self):
         self.pyme = get_object_or_404(PYME, user=self.request.user)
-        return Autoevaluation.objects.filter(pyme=self.pyme).order_by('last_time_edition')
+        return Autoevaluation.objects.filter(pyme_id=self.pyme).order_by('last_time_edition')
 
       
 class ResultDetail(LoginRequiredMixin, DetailView):
+    """View to handles the detail of the autoevalution ressult
```
Comment (jandrovins) [inline mm_evaluation/views.py:188]: This does not comply Google's pydoc strings conventions.

File: mm_evaluation/views.py (hunk 3 above)
Comment (jandrovins) [reply mm_evaluation/views.py:264]: You should not yell the future us debugging this code :laughing:

File: mm_evaluation/models.py (hunk 7 above)
Comment (clopezr9) [reply mm_evaluation/models.py:46]: I'm pretty sure that I did not do that

File: diagnospymes/settings.py (hunk 6 above)
Comment (clopezr9) [reply diagnospymes/settings.py:60]: It is not. Corrected

**Comment by clopezr9** [comment]
I think its ready

**Pull request merged**